azure/compute/src/main/java/org/jclouds/azurecompute/compute/strategy/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,326 @@
+        this.deploymentResponseToVirtualMachine = checkNotNull(deploymentResponseToVirtualMachine, "deploymentResponseToVirtualMachine");
+        this.serviceResponseToVirtualMachine = checkNotNull(serviceResponseToVirtualMachine, "serviceDeploymentResponseToDeployment");
+        this.osImageListResponseToOSImage = checkNotNull(osImageListResponseToOSImage, "osImageListResponseToOSImage");
+        this.osImageResponseToOSImage = checkNotNull(osImageResponseToOSImage, "osImageResponseToOSImage");
+        this.groupNamingConvention = namingConvention;
+//      this.serverLatestJobCompleted = retry(new ServerLatestJobCompleted(client.getJobServices()),
+//            timeouts.nodeRunning * 9l / 10l);
+//      this.serverLatestJobCompletedShort = retry(new ServerLatestJobCompleted(client.getJobServices()),
+//            timeouts.nodeRunning * 1l / 10l);
+    }
+
+
+    @Override
+    public NodeAndInitialCredentials<VirtualMachine> createNodeWithGroupEncodedIntoName(String group, String name,
+                                                                                    Template template) {
+        logger.info("createNodeWithGroupEncodedIntoName");
+        checkNotNull(template, "template was null");
+        checkNotNull(template.getOptions(), "template options was null");
+        checkArgument(template.getOptions().getClass().isAssignableFrom(AzureTemplateOptions.class),
+                "options class %s should have been assignable from AzureTemplateOptions", template.getOptions().getClass());
+        AzureTemplateOptions templateOptions = template.getOptions().as(AzureTemplateOptions.class);
+        checkNotNull(templateOptions.getStorageAccount(), "storage account should be specified in order to create node");
+
+        // check if hosted service exists
+        HostedServiceGetDetailedResponse hostedService = client.getHostedServiceApi().getHostedServiceWithDeployments(group);
+        if (hostedService == null) {
+            HostedServiceCreateParameters parameters = new HostedServiceCreateParameters();
+            parameters.setServiceName(group);
+            parameters.setLocation(template.getLocation().getId());
+            client.getHostedServiceApi().createHostedService(parameters);
+        }
+
+        //required
+        ArrayList<ConfigurationSet> configlist = new ArrayList<ConfigurationSet>();
+        ConfigurationSet configset = new ConfigurationSet();
+        configset.setConfigurationSetType(template.getImage().getOperatingSystem().getFamily() == OsFamily.LINUX ?
+                ConfigurationSetTypes.LINUXPROVISIONINGCONFIGURATION : ConfigurationSetTypes.WINDOWSPROVISIONINGCONFIGURATION);
+//        configset.setConfigurationSetType(ConfigurationSetTypes.WINDOWSPROVISIONINGCONFIGURATION);
+        String vmName = name.substring(group.length() + 1);
+//        required
+        configset.setComputerName(vmName);
+        String adminuserPassword = "Pamir5463";
+        String adminUserName = "farlen";
```

Comment:
This can't be hardcoded here and must be read from the template options, as it provides fields to override the login username, password etc. Default credentials can be set in the image transformation function or as default properties in the api metadata, where you can specify credentials for each os family.